resource_estimator/src/counts.rs

```diff
@@ -37,6 +37,12 @@ pub struct LogicalCounter {
     /// Global allocation barrier (when calling global barrier this is advanced
     /// to allocate new qubits after the barrier)
     allocation_barrier: usize,
+    /// If set, enables more precise volume computation: in this mode,
+    /// un-allocated qubits do not contribute to the logical volume, as they do
+    /// not need to be error corrected. The vector contains for each qubit the
+    /// data to compute the logical depth collected while the qubit was
+    /// allocated.
+    precise_volume: Option<Vec<(VolumeEntry, VolumeEntry)>>,
```

Comment:
The tuple here hides meanings of its elements. This is especially dangerous because types are the same. Could we please have a struct with clearly named volume entries instead?